src/misc/ansi_code_markup.cpp

```diff
@@ -25,97 +25,213 @@
+	 * and info().group == TagGroup::Erasers
+	 *
+	 * \return const EraseInfo&
+	 */
+	const EraseInfo &erase_info() const { return e_info; }
 
 private:
-	ColorParser color;
-	char ansi_code[10] = {0};
-	const std::unordered_map<std::string, int> tags = {
-	        {"reset", 0}, {"b", 1},       {"dim", 2},
-	        {"i", 3},     {"u", 4},       {"s", 9},
-	        {"blink", 5}, {"inverse", 7}, {"hidden", 8},
+	bool parse_color_val(const std::string &val);
+	bool parse_erase_val(const std::string &val);
+
+	bool is_closed = false;
+	bool is_valid = false;
+
+	TagInfo t_info;
+	ColorInfo c_info;
+	EraseInfo e_info;
+
+	static inline const std::string light_prefix = "light-";
+
+	static inline const std::unordered_map<std::string, TagInfo> tags = {
+	        {"color", {TagGroup::Colors, TagName::Color, -1}},
+	        {"bgcolor", {TagGroup::Colors, TagName::BGColor, -1}},
+	        {"erasel", {TagGroup::Erasers, TagName::EraseL, -1}},
+	        {"erases", {TagGroup::Erasers, TagName::EraseS, -1}},
+	        {"i", {TagGroup::Styles, TagName::It, 3}},
+	        {"b", {TagGroup::Styles, TagName::Bold, 1}},
+	        {"u", {TagGroup::Styles, TagName::Ul, 4}},
+	        {"s", {TagGroup::Styles, TagName::Strike, 9}},
+	        {"blink", {TagGroup::Styles, TagName::Blink, 5}},
+	        {"dim", {TagGroup::Styles, TagName::Dim, 2}},
+	        {"hidden", {TagGroup::Styles, TagName::Hidden, 8}},
+	        {"inverse", {TagGroup::Styles, TagName::Inverse, 7}},
+	        {"reset", {TagGroup::Misc, TagName::Reset, 0}},
+	};
+
+	static inline const std::unordered_map<std::string, ColorInfo> color_values = {
+	        {"black", {ColorName::Black, 30, false}},
+	        {"red", {ColorName::Red, 31, false}},
+	        {"green", {ColorName::Green, 32, false}},
+	        {"yellow", {ColorName::Yellow, 33, false}},
+	        {"blue", {ColorName::Blue, 34, false}},
+	        {"magenta", {ColorName::Magenta, 35, false}},
+	        {"cyan", {ColorName::Cyan, 36, false}},
+	        {"white", {ColorName::White, 37, false}},
+	        {"default", {ColorName::Default, 39, false}},
+	        {"light-black", {ColorName::Black, 30, true}},
+	        {"light-red", {ColorName::Red, 31, true}},
+	        {"light-green", {ColorName::Green, 32, true}},
+	        {"light-yellow", {ColorName::Yellow, 33, true}},
+	        {"light-blue", {ColorName::Blue, 34, true}},
+	        {"light-magenta", {ColorName::Magenta, 35, true}},
+	        {"light-cyan", {ColorName::Cyan, 36, true}},
+	        {"light-white", {ColorName::White, 37, true}},
+	        {"light-default", {ColorName::Default, 39, true}},
+	};
+
+	static inline const std::unordered_map<std::string, EraseInfo> eraser_extents = {
+	        {"end", {EraseExtents::End, 0}},
+	        {"begin", {EraseExtents::Begin, 1}},
+	        {"entire", {EraseExtents::End, 2}},
 	};
 };
 
-static TagParser tag_parser;
+Tag::Tag(std::string &tag, std::string &val, const bool close)
+        : t_info(),
+          c_info(),
+          e_info()
```

Comment:
Suggest moving these to initializers:

``` c++
	TagInfo t_info = {};
	ColorInfo c_info = {};
	EraseInfo e_info = {};
```

Then these are good for all derived constructors too.